appdaemon/plugins/hass/hassplugin.py

```diff
@@ -373,6 +373,10 @@ async def get_updates(self):  # noqa: C901
                         else:
                             await self.evaluate_started(False, self.hass_booting)
                     else:
+                        metadata = result["event"].copy()
+                        metadata.pop("event_type")
+                        metadata.pop("data")
+                        result["event"]["data"]["metadata"] = metadata
```

Comment:
Thanks for the suggestion. My code is almost the same but instead of popping the keys from `result["event"]` I made a copy of the dict and then modified it. Now I applied your suggestion as it will give the same result.
As for checking for `state_changed` event type I would like to know why we should exclude this as moving these keys to metadata would not affect the `state_changed` event, unless I am missing something here.

Reply:
Ok if same that's fine, as I am not able to test it so not sure what was what; though I noticed you updated your description and it was what was envisioned. I wanted to avoid state change, as I felt it had the data you wanted, but I am not fixated on avoiding it, and if it improves on what you want, I am all for it, as long as it doesn't break any code.

@acockburn when you available, can you kindly give your input on this, as that we can move into merging this pending when you want to release.

Kind regards

Reply:
Ah and looking at your original code again, yeah you right. I must have been high on something if I didn't notice it does the same thing

Reply:
DONE